source/1_engine/rex_engine/include/rex_engine/string/stringid.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "rex_engine/string/stringentry.h"
+#include "rex_engine/types.h"
+#include "rex_std/ostream.h"
+#include "rex_std/string.h"
+
+namespace rex
+{
+  class StringID
+  {
+  public:
+    StringID();
+    explicit StringID(rsl::string_view characters);
+
+    explicit operator u32() const;
+
+    bool operator==(const StringID& other) const;
+    bool operator!=(const StringID& other) const;
+
+    bool operator==(const StringEntryID& entryID) const;
+    bool operator!=(const StringEntryID& entryID) const;
+
+    rsl::string_view to_string_view() const;
+    bool is_none() const;
+    u32 value() const;
+
+  private:
+    /** Index into the StringID array (used to find String portion of the string/number pair used for comparison) */
+    StringEntryID m_comparison_index;
+  };
+
+  StringID create_sid(rsl::string_view stringView);
```

Comment:
there is no way to remove string_ids that is true but I don't think this is required though. what might be useful is a clear or something. for instance, if we switch between different states of the application it might be an idea to free up some memory for strings that will not be used. however, we do need to keep in mind if we switch back to the other state the string pool has to be filled again (unless we store it somewhere on disk)

Reply:
I think it's worth to keep in mind but doesn't have to be implemented in this change